src/games/rps/index.ts

```diff
@@ -178,7 +179,7 @@ export const getRockPaperScissorsResult = async ({
     let rawResults: RPSParsedJson[] = [];
     let txDigests: string[] = [];
 
-    while (results.length === 0) {
+    while (results.length < gameInfos.length) {
```

Comment:
Does this condition check for if the results are duplicated results?

Reply:
In row 182 it didn't check if the results are duplicated or not.

But within this while loop, in row 195

```typescript
events.data.forEach((event) => {
      const { bet_id, settlements } = event.parsedJson as RPSParsedJson;

      if (bet_id == gameInfos[results.length].gameId) { //<- row 195
      ....
      }
    }, []);
```
This is how we collect the results from events.

The code will pick a result that has the same gameId of gameInfo, and will automatically change to find another gameId in the gameInfos array.

The automatic mechanism works with the length of the results array.

So even if we got a duplicated result, it won't be collected in our results array.

Unless we get the same gameId in the gameInfos array....., I haven't met this situation yet